arch/stream_traces.py

```diff
@@ -29,14 +30,10 @@ def process_log_line(line):
         logging.error(f"Failed to send trace to otel-tracing: {e}")
 
 
-with open(envoy_log_path, "r") as f:
-    # Seek to the end of the file so we only read new lines
-    f.seek(0, os.SEEK_END)
-    while True:
-        line = f.readline()
-        if not line:
-            time.sleep(1)
-            continue
-        tokens = line.split("prompt_gateway: upstream_llm trace details: ")
-        if len(tokens) > 1:
-            process_log_line(tokens[1])
+for line in sys.stdin:
+    if not line:
+        time.sleep(1)
```

Comment:
why we need sleep, this will make application very slow

Reply:
that is only when line is empty otherwise it will block forever